xc7/fasm2bels/clb_models.py

```diff
@@ -209,57 +250,101 @@ def cleanup_slice(top, site):
 
     """
     carry4 = site.maybe_get_bel('CARRY4')
+    if carry4 is not None:
 
-    if carry4 is None:
-        return
+        # Simplest check is if the CARRY4 has output in used by either the OUTMUX
```

Comment:
Let's refactor the cleanup into "cleanup_carry4", "cleanup_srl", assuming that no logic/data is shared between them.

Reply:
Done.